### PR TITLE
=osgi #19323 Add to documentation on OSGi

### PR DESCRIPTION
**Pull request by briantopping**

### CODE REVIEW

**Comment by johanandren** [comment]
Well written text! I'm however not sure it belongs in the Akka docs. 
To me this feels more like a text on a site promoting OSGi and I am 
not sure we are interested in doing that?

File: akka-docs/rst/additional/osgi.rst [hunk 1]
```diff
@@ -1,12 +1,100 @@
...
+concepts to learn in the JRE and the added value to teams that were managing well with straight J2EE was not obvious.
+Facilities emerged to "wrap" binary JARs so they could be used as bundles, but this functionality was only used in limited
+situations. An application of the "80/20 Rule" here would ahve that "80% of the complexity is with 20% of the configuration",
+but it was enough to give OSGi a reputation that has stuck with it to this day.
+
+Today, the situation is much different. Most popular artifacts available in open source repositories are instrumented
+with OSGi bundle metadata and are tested and deployed in production environments. Generating the metadata is a reduced
+to a simple build step with permissive defaults via plugins that are available for all popular build tools. Various
+deployment options exist for embedding OSGi in the smallest Java applications (such as for plugin architectures) all
+the way up to enterprise containers with modular (of course!) command lines and comprehensive management interfaces.
```
Comment (drewhk) [inline akka-docs/rst/additional/osgi.rst]: I am not sure this and the above paragraph belongs to Akka documentation. It is too much opinion in there that is outside of the scope of Akka IMHO.

**Comment by drewhk** [comment]
Apart from the introductory text (which for me feels a bit like selling OSGi) it is a great addition!

**Comment by johanandren** [comment]
Isn't the other parts generic for OSGi and unrelated to Akka as well? (However well written they are)

**Comment by briantopping** [comment]
Hi guys, thanks for the great feedback! I'm happy to adjust this. I'm not advocating on the part of OSGi (or not trying to anyway). 

The intent with writing the doc the way I did was that @rkuhn and others said that he didn't know enough about it. Since "reading lists" don't get read, I tried to distill it down to the most important elements here. What are the minimums that someone needs to know to get beyond "this thing sucks!". That's usually a result of not knowing these basics.

The fact that akka-osgi project exists promotes OSGi, just as akka-clustering promotes clustering and akka-persistence promotes using a database. In it's current state, it's just a waste of space, IMHO. Nobody knows how to use it or support it and seems very frustrating for almost everyone.

**Comment by briantopping** [comment]
Based on the feedback here so far, I replaced the paragraph that @drewhk pointed out with one that tries to tie in why the document is here and a reference for a doc that can go deeper. Hopefully it gives a better feel overall, but it's hard for me to tell as author. What do you guys think?

**Comment by rkuhn** [comment]
@briantopping sorry for not responding to these, we are currently heads down working on getting master to build with Streams & HTTP; should be done tomorrow

**Comment by briantopping** [comment]
Sorry @rkuhn and @drewhk, I should have noted that I know you guys are busy with that. No rush at all, next week fine with me.

**Comment by briantopping** [comment]
I pooched this branch, please see #19435.
